### PR TITLE
Library refresh

### PR DESCRIPTION
**Pull request by ed-xmos**
- [x]  Remove old examples
- [x]  Examples XCCM
- [x]  Docs (xmosdoc)
- [x]  Jenkins
- [x]  lib_checks
- [x]  tests to build using XCCM
- [x]  tests update to testsupport
- [x]  New lib style
- [ ] AN docs (FUTURE PR)
- [x] Fix compiler warnings
- [ ] Add XS3 support to tests  (FUTURE PR)
- [ ] Review and fix issues as appropriate  (FUTURE PR)
- [ ] Hardware regression test?  (FUTURE PR)
- [ ] Remove xcommon for example but add legacy build test (FUTURE PR)

### CODE REVIEW

File: tests/test_time_rx_tx.py [hunk 1]
```diff
@@ -64,45 +74,97 @@ def packet_checker(packet, phy):
...
+    lock_path = f"{testname}.lock"
+    # xdist can cause race conditions so use a lock
+    with FileLock(lock_path):
+        result = subprocess.run('cmake -B build -G "Unix Makefiles"', shell=True, cwd=testname)
+        result.check_returncode()
+        result = subprocess.run(f'xmake -j 8 -C build {config}', shell=True, cwd=testname)
+        result.check_returncode()
+
+
+# I had problems with the comparison tester putting a newline in the expected regex so wrote custom one and replaced create_expect
```
Comment (shuchitak) [inline tests/test_time_rx_tx.py:105]: If this is fixing a general problem, maybe we upstream it into the test_support repo

**Review by shuchitak — CHANGES_REQUESTED** [review]
Requested changes related to removing the ADAT references from README, looks good otherwise.

File: README.rst [hunk 2]
```diff
@@ -1,14 +1,33 @@
-Ethernet MAC library
-====================
+:orphan:
 
+########################
+lib_adat: ADAT lightpipe
```
[the remark this answers is not in the document]
Comment (ed-xmos) [reply README.rst:4]: Ooops - fixed

File: tests/test_time_rx_tx.py (hunk 1 above)
Comment (ed-xmos) [reply tests/test_time_rx_tx.py:105]: I didn't get to the bottom of it. It was only that test and took ages to figure out a workaround. I'll stick it on the list for later as it's not a show stopper.

File: examples/AN00199_gigabit_ethernet_demo_explorerkit/Makefile [hunk 3]
```diff
@@ -25,11 +25,11 @@ APP_NAME =
 
 XCC_FLAGS = -g -report -DDEBUG_PRINT_ENABLE -save-temps -Os -Xmapper --map -Xmapper MAPFILE -DRGMII=1
 
-XCC_FLAGS_icmp = $(XCC_FLAGS) -Wno-reinterpret-alignment
+XCC_FLAGS_icmp.xc = $(XCC_FLAGS) -Wno-reinterpret-alignment
```
[the remark this answers is not in the document]
Comment (ed-xmos) [reply examples/AN00199_gigabit_ethernet_demo_explorerkit/Makefile:28]: Hmm, thinking about it maybe not. But if I remove this we should have a legacy build test to make sure xcommon works.

**Review by shuchitak — CHANGES_REQUESTED** [review]
Requested changes about the rst heading level. Looks good otherwise!

**Review by shuchitak — APPROVED** [review]

**Pull request merged**